src/Voron/Data/BTrees/TreeCursor.cs

```diff
@@ -46,78 +24,45 @@ public void Dispose()
         // The bulk of the clean-up code is implemented in Dispose(bool)
         private void Dispose(bool disposing)
         {
-            if (disposing)
-            {
-                _state.Clear();
-                _pagesByNumPool.Free(_state);
-            }
+            if (disposing == false) 
+                return;
+
+            _statePages.WeakClear();
```

Comment:
What is the motivation for using `WeakClear()` here? It will keep referencing existing objects in the underlying array.

Reply:
`TreeCursor` is pretty small and we are paying a lot of time for clearing it (too many repetitions happening too often). The tradeoff here is we are OK to keep some garbage to avoid having to spend the time.